joomla/joomla-cms
# [4.0] Add the Report-To header to the http header config and allow to set the script-dynamic header

### Summary of Changes

Add the [Report-To](https://w3c.github.io/reporting/) ([#2](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to)) header to the http header config and allow to set the [script-dynamic](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src) header 

### Testing Instructions

- go to the plg_system_httpheaders
- notice there is no report-to header in the `force header` option.
- apply this patch
- the option is there
- set the option to any value. ([or see here the details](https://w3c.github.io/reporting/) + [example](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to))
- revert the patch
- go to the com_csp options
- notice there is no strict-dynamic setting
- apply this patch
- notice there is now a strict dynamatic setting. [(details)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src)
- enable the strict dynamic setting
- (when in custom mode make sure you setup a inital script-src dummy rule)
- and check the generated csp header. (F12 -> Network tab)

### Expected result

- you can set the report-to via the plugin
- you can set the strict-dynamic option for the CSP

### Actual result

- you can not set the report-to via the plugin
- you can not set the strict-dynamic option for the CSP

### Documentation Changes Required

The script-dynamic options needs to be added to doc pages the Report-To has been added already
https://docs.joomla.org/index.php?title=Help4.x:Components_CSP_Reports_Options
https://docs.joomla.org/J4.x:Http_Header_Management
